lib/web_ui/dev/utils.dart

```diff
@@ -328,6 +334,32 @@ mixin ArgUtils<T> on Command<T> {
 
   /// Extracts a string argument from [argResults].
   String stringArg(String name) => argResults![name] as String;
+
+  RuntimeMode get runtimeMode {
+    final bool isProfile = boolArg('profile');
+    final bool isDebug = boolArg('debug');
+    if (isProfile && isDebug) {
+      throw ToolExit('Cannot specify both --profile and --debug at the same time.');
+    }
+    if (isProfile) {
+      return RuntimeMode.profile;
+    } else if (isDebug) {
+      return RuntimeMode.debug;
+    } else {
+      return RuntimeMode.release;
+    }
+  }
+}
+
+io.Directory getBuildDirectoryForRuntimeMode(RuntimeMode runtimeMode) {
+  switch (runtimeMode) {
```

Comment:
you can use the new switch expression feature here!

=> switch (runtimeMode) { RuntimeMode.debug => envi.wasmdebugOUt, ...}

Reply:
that should be same same. Weird!

Reply:
Ah, I had the syntax slightly wrong. It worked when I copied your example more closely. Thanks, this is slick!